### ISSUE TITLE
Apostrophes get treated as quoted text

### ISSUE THREAD

**Issue by fcastilloec** [goto-bus-stop/get-artist-title#40]
When fixing #7 it introduced a bug when a single apostrophe will cause it to not parse any artist and title.

As an example of what should work:
```js
const getArtistTitle = require('get-artist-title')
getArtistTitle("Sinéad O'Connor - Nothing Compares 2 U")
//→ ["Sinéad O'Connor", 'Nothing Compares 2 U']
```
Instead, I get `undefined`.

Also, this bug only happens if the apostrophe is in the artist part. If only the title has it, there's no issue since the code already found the separator before finding a quote character.

Another example, I had to find a song from her that contained an apostrophe:
```js
const getArtistTitle = require('get-artist-title')
getArtistTitle("Sinéad O'Connor - The Emperor's New Clothes")
//→ ["Sinéad O'Connor", "The Emperor's New Clothes"]
```
Instead, I get `[ 'Sinéad O', 'Connor - The Emperor' ]`.

I think that apostrophes should be removed from the list of [`toggleChars`](https://github.com/goto-bus-stop/get-artist-title/blob/default/lib/plugins/base.js#L77). This is an issue mostly with Western languages, including all European alphabets (Greek, Latin, Cyrillic, etc), so I'm not sure how this will affect other alphabets/languages.

**Comment by goto-bus-stop** [comment]
Yeah I've also noticed this a bit. I think I agree it should be removed.

Maybe it can do something else to more specifically address the common "ARTIST 'Song Name' (some cruft)" pattern that's common in Korean music (that's what I mostly used the package for myself at the time, and why IIRC that's why the quote characters added in the first place).

**Comment by fcastilloec** [comment]
I have a suggestion that won't solve all cases, maybe you can check if there's a single apostrophe on the string (as with the first example I provided), if you find two (or more) then you can add it to toggleChars. As an example:

```js
function splitArtistTitle (str) {
  var aposAsQuotes = str.split('\'').length >= 3 // Check if there's more than one apostrophe
  ... // remaining code
  if (idx > -1 && !inQuotes(str, idx, aposAsQuotes)) {
    .. // remaining code
  }
}
function inQuotes (str, idx, aposAsQuotes) {
  var toggleChars = aposAsQuotes ? '"\'' : '"'
  ... // remaining code
}
```
This won't help in cases where the artist and song both have apostrophes, but I think those cases are not as common and we might live with it.

Another idea is to check for Asian characters in the string, and similar to the previous code, add apostrophes to `toogleChars` if that's the case. For any European alphabet characters, we disregard it. This will hopefully fix both examples I provided and keep working in the case you mentioned about Korean music.
I wouldn't really know how to do this, but perhaps this post could help: https://stackoverflow.com/a/43419070/4868556
